### ISSUE TITLE
Remove "auto-naming" of copied blocks

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#1388]
in ref to #1383 discussion on auto-naming copied blocks in SE, its sugested that this is dropped, so copies only will retain user edited names. Defaults will be shown as blanc /empty-string

**Comment by curlymorphic** [comment]
The blanking of pattern name, is that to only happen if copying to another track, or if copying to same track aswell?

**Comment by Sti2nd** [comment]
I must take if for granted you mean when it is the default name (name of the instrument). Only between different tracks. No point in deleting the default name copied in the same track as it doesn't show anyway, and as Tres explains it can be useful that the instrument name is there.

I wonder if it would make sense that the copied patterns take the name of the instrument you are copying... Actually is has to be that way, cause a pattern must have a name. That LMMS probably does automatically with empty pattern names.

**Comment by tresf** [comment]
> The blanking of pattern name, is that to only happen if copying to another track, or if copying to same track aswell?

Hard to explain so I'll give some use-case examples.

Scenario 1: Don't use the default name.

 * First instrument track called "TripleOscillator"
 * Second instrument track called "Nescaline"
 * Make new Piano Roll segment in TripleOscillator
 * Copy to Nescaline, it now has the name "TripleOscillator".  This is **NOT** desired.

---

Scenario 2:  Do use the custom name.

 * First instrument track called "TripleOscillator"
 * Second instrument track called "Nescaline"
 * Make new Piano Roll segment in TripleOscillator
 * Name this Piano Roll segment "Main Melody"
 *  * Copy to Nescaline, it now has the name "Main Melody".  This **IS** desired.



So if the pattern name matches the track name, don't copy it.  If the pattern name is different from the track name, copy it. :+1:

**Comment by Lukas-W** [comment]
Is there any use for the default track name, i.e. wouldn't it be cleaner to make default names empty?

**Comment by tresf** [comment]
> Is there any use for the default track name, i.e. wouldn't it be cleaner to make default names empty?

I believe it is fine either way.  Empty is probably slightly better from a usability perspective. i.e.
Preset is called "LowBass.xpf".  Instrument track is called "LowBass".  Pattern is called "LowBass".
 * I rename it to "LowBass Pattern1", I've saved a few keystrokes by having the name pre-populated
 * I rename it to "Main Bassline", I had to hit backspace first.

-Tres
